ksqldb-engine/src/main/java/io/confluent/ksql/internal/PullQueryExecutorMetrics.java

```diff
@@ -550,6 +525,28 @@ private Map<MetricsKey, Sensor> configureLatencySensorMap() {
     });
   }
 
+  public void configureThreadPoolMetrics() {
+    final MetricName coordinatorThreadsAvailable = metrics.metricName(
+        PULL_REQUESTS + "-coordinator-thread-pool-free-size",
+        ksqlServicePrefix + PULL_QUERY_METRIC_GROUP,
+        "Number of available threads in the coordinator pool",
+        customMetricsTags
+    );
+    if (!metrics.metrics().containsKey(coordinatorThreadsAvailable)) {
+      metrics.addMetric(coordinatorThreadsAvailable, coordinatorThreadPoolSizeGauge);
```

Comment:
Can't you just directly get the supplier coordinatorThreadPoolSupplier and just call `metrics.addMetric(coordinatorThreadsAvailable, (Gauge<Integer>) (config, now) -> coordinatorThreadPoolSizeGauge.get())`.  I think that would allow you to skip a lot of the complexity, including have to report for every request.

Reply:
I tried doing it like this initially but the values of the metric were not measured correctly. I tried now again, and it works so I must have been doing something else wrong. Anyway, this change simplifies the code a lot so thanks! :)